terraform.md

```diff
@@ -0,0 +1,120 @@
+# Terraform
+
+Terraform is a really powerful tool with a lot of freedom in the way you use it.
+It isn't a very opinionated tool, however if you don't stick with rules it can
+easily get complicated and hard to maintain.
+
+This is why we decided to write our own guideline on how to structure a
+`terraform` codebase.
+
+## File Structure and Filenames
+
+Files should be named with the AWS (or other cloud provider) components they are
+referring to. Keeping this open to future needs, if a component is getting more
+in use than others we can make the filename more specific.
+The idea is to keep a balance between too many small files and too little huge
+files.
+
+```
+/my-projet/
```

Comment:
To make things clearer internally, we shoudl mirror our own desired folder structure - so something like:
```
modules/
    database/
        ec2.tf
        ...
workspaces
    someclient-prod/
        main.tf  # if we want to keep this in the workspace
```